Add Feature Preview documentation (#846)
gitpod-io/website: src/docs/feature-preview.md

# Feature Preview

You can enable Feature Preview from the Settings. This will enable a beta preview of some of the latest functionalities and features that are being actively developed for your user account. Some of them are user-facing features while others take place in the backstage.

You can disable Feature Preview at any time.

If you have suggestions on how we can improve a feature, please provide feedback by [opening an issue](https://github.com/gitpod-io/gitpod/issues/new/choose).

> > Once you enable this, existing and new workspaces will automatically start with all beta features. Disabling this could make existing workspaces unstable. 

Author: **thought**: If this sounds inaccurate or intimidating we could also remove this section altogether.

Reviewer: sounds good to me

> ## Root Access

Author: **question**: I've added _Root Access_ here based on the [relevant discussion](https://typefox.slack.com/archives/CGVQAPNPL/p1605602104426700) (internal). We can mention in the description below any details or references to running Docker, etc. Feel free to suggest any alternatives.

> Root Access allows you to attain privileged control within a workspace and be able to run commands as root as well as run Docker and Docker Compose.

Author: **question**: Not sure if this description is accurate.

Reviewer: Root Access allows you to attain privileged control within a workspace and be able to run commands as root.

This feature also enables Docker in your workspace. Run `sudo docker-up` to start a Docker daemon.

Author: ```suggestion
Root Access allows you to attain privileged control within a workspace and be able to run commands as root.

This feature also enables Docker in your workspace. Run `sudo docker-up` to start a Docker daemon.
```

## Default IDE

Gitpod uses Theia as the default IDE. This setting allows you to choose between [Theia](https://github.com/eclipse-theia/theia) and [Code](https://github.com/microsoft/vscode) editors as well as provide a custom IDE image.

You can choose between 3 options:

- Theia
- Code

> - Custom

Author: **question**: The custom image option seems to be missing in the preview environment. Is this still an option?

Author: Renaming this to _Image_ to match the label on staging. /cc @akosyakov 

```suggestion
- Image
```

# Feature Preview

> You can enable Feature Preview from the Settings. This will enable a beta preview of some of the latest functionalities and features that are being actively developed for your user account. Some of them are user-facing features while others take place in the backstage.

Author: Removing a small piece that was originally added while having _Teams_ feature in mind. We can always add this back later.

```suggestion
You can enable Feature Preview from the Settings. This will enable a beta preview of some of the latest functionalities and features that are being actively developed. Some of them are user-facing features while others take place in the backstage.
```
